File: 백준/Silver/14916. 거스름돈/거스름돈.java
import java.io.BufferedReader;
import java.io.BufferedWriter;
import java.io.InputStreamReader;
import java.io.OutputStreamWriter;
import java.util.*;
import java.io.IOException;

public class Main {
    public static void main(String[] args) throws IOException, InterruptedException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(System.out));

        int n, output;

        n = Integer.parseInt(br.readLine());
        if(n == 1 || n == 3)
            output = -1;
        else {
            output = n / 5;
            n -= output * 5;
            if(n % 2 != 0) {
                n += 5;
                output--;
            }
            output += n / 2;
        }
        bw.write(output + "");
        bw.flush();
    }
}

File: 백준/Bronze/5585. 거스름돈/거스름돈.kt
import java.io.BufferedReader
import java.io.BufferedWriter
import java.io.InputStreamReader
import java.io.OutputStreamWriter
import java.util.Arrays
import java.util.StringTokenizer

fun main(args: Array<String>) {
    var br = BufferedReader(InputStreamReader(System.`in`))
    var bw = BufferedWriter(OutputStreamWriter(System.out))

    val input = Integer.parseInt(br.readLine())
    var money = 1000 - input
    var count = 0;
    val coinArr = arrayOf(500, 100, 50, 10, 5, 1)
    for(i in 0 until coinArr.size) {
        count += money / coinArr[i]
        money %= coinArr[i]
    }
    bw.write("$count")
    bw.flush()
}